mmv1/third_party/terraform/tests/resource_eventarc_channel_test.go.erb

```diff
@@ -0,0 +1,214 @@
+<% autogen_exception -%>
+package google
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	dcl "github.com/GoogleCloudPlatform/declarative-resource-client-library/dcl"
+	<% if version == "ga" -%>
+	eventarc "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/eventarc"
+	<% else -%>
+	eventarc "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/eventarc/beta"
+	<% end %>
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccEventarcChannel_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"region":        getTestRegionFromEnv(),
+		"project_name":  getTestProjectFromEnv(),
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckEventarcChannelDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEventarcChannel_basic(context),
+			},
+			{
+				ResourceName:      "google_eventarc_channel.primary",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccEventarcChannel_cryptoKeyUpdate(t *testing.T) {
+	t.Parallel()
+
+	region := getTestRegionFromEnv()
+	key1 := BootstrapKMSKeyWithPurposeInLocationAndName(t, "ENCRYPT_DECRYPT", region, "tf-bootstrap-key1")
+	key2 := BootstrapKMSKeyWithPurposeInLocationAndName(t, "ENCRYPT_DECRYPT", region, "tf-bootstrap-key2")
+
+	context := map[string]interface{}{
+		"region":        region,
+		"project_name":  getTestProjectFromEnv(),
+		"key_ring":      GetResourceNameFromSelfLink(key1.KeyRing.Name),
+		"key1":          GetResourceNameFromSelfLink(key1.CryptoKey.Name),
+		"key2":          GetResourceNameFromSelfLink(key2.CryptoKey.Name),
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckEventarcChannelDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEventarcChannel_setCryptoKey(context),
+			},
+			{
+				ResourceName:      "google_eventarc_channel.primary",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccEventarcChannel_cryptoKeyUpdate(context),
+			},
+			{
+				ResourceName:      "google_eventarc_channel.primary",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccEventarcChannel_basic(context map[string]interface{}) string {
+	return Nprintf(`
+data "google_project" "test_project" {
+	project_id  = "%{project_name}"
+}
+
+resource "google_eventarc_channel" "primary" {
+	location = "%{region}"
+	name     = "tf-test-name%{random_suffix}"
```

Comment:
This is inconsistent with the documentation on the `name` field, which states a longer format must be used:

```
* `name` -
  (Required)
  Required. The resource name of the channel. Must be unique within the location on the project and must be in `projects/{project}/locations/{location}/channels/{channel_id}` format.
```

Reply:
Acknowledged

Reply:
I am updating the docs in the DCL to remove the longer format